### ISSUE TITLE
qat_hw_chachapoly.c: null check implies pointer is null, but it has been dereferenced before the null check

### ISSUE THREAD

**Issue by ColinIanKing** [intel/QAT_Engine#269]
In function qat_chacha20_poly1305_init there is a null check on cp_ctx->opd (implying it may be null) in the following code:

```
init_err:
    if (cp_ctx->opd->pIv)                    
        qaeCryptoMemFreeNonZero(cp_ctx->opd->pIv); 
    if (cp_ctx->opd)
        OPENSSL_clear_free(cp_ctx->opd, sizeof(template_opData));
    if (cp_ctx->session_data != NULL) {
        OPENSSL_free(cp_ctx->session_data);
        cp_ctx->session_data = NULL;
    }                          
    return ret;
```

however cp_ctx->opd accessed in the function before the null check, in the if statement before it, so this may trip an null pointer dereference issue.

```
init_err:
    if (cp_ctx->opd->pIv)                    
        qaeCryptoMemFreeNonZero(cp_ctx->opd->pIv); 
```

**Comment by Yogaraj-Alamenda** [comment]
@ColinIanKing Thanks for the inputs. We will fix it in upcoming release

**Comment by Yogaraj-Alamenda** [comment]
Fixed in the commit 418411b5206c2727f6578fbca9dbd11df98a237d
